.github/workflows/cd-deploy.yml

```
@@ -1,73 +1,242 @@
+  # ──────────────────────────────────────────────────────────────
+  # JOB 1 — Provision / update AWS infrastructure via Pulumi
+  #         Outputs the bucket/EB/ECR names needed by later jobs.
+  # ──────────────────────────────────────────────────────────────
+  deploy-infra:
+    name: Pulumi — Provision Infrastructure
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write   # Required for OIDC → AWS
+      contents: read
+    outputs:
+      frontend-bucket:   ${{ steps.pulumi.outputs.FrontendBucketName }}
```

actionlint (1.7.12)

[error] property "frontendbucketname" is not defined in object type {output: string}

(expression)